flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/DynamicFilteringData.java

```diff
@@ -26,30 +26,49 @@
 import org.apache.flink.core.memory.DataInputViewStreamWrapper;
 import org.apache.flink.table.api.TableException;
 import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
 import org.apache.flink.table.types.logical.RowType;
 
 import java.io.ByteArrayInputStream;
 import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 import static org.apache.flink.util.Preconditions.checkNotNull;
 
 /** Data for dynamic filtering. */
 public class DynamicFilteringData implements Serializable {
+
+    public static final Set<LogicalTypeRoot> SUPPORTED_TYPES;
+
+    static {
+        Set<LogicalTypeRoot> supported = new HashSet<>();
+        supported.add(LogicalTypeRoot.INTEGER);
+        supported.add(LogicalTypeRoot.BIGINT);
+        supported.add(LogicalTypeRoot.SMALLINT);
+        supported.add(LogicalTypeRoot.TINYINT);
+        supported.add(LogicalTypeRoot.VARCHAR);
+        supported.add(LogicalTypeRoot.CHAR);
+        SUPPORTED_TYPES = Collections.unmodifiableSet(supported);
```

Comment:
I think date/time/timestamp can be supported

Reply:
DATE, TIME_WITHOUT_TIME_ZONE, TIMESTAMP_WITH_TIME_ZONE, TIMESTAMP_WITH_LOCAL_TIME_ZONE, TIMESTAMP_WITHOUT_TIME_ZONE are added. 
In fact as the field values are acquired and used to form a GenericRowData in Hive source in the same way as in the collector operator, I suppose all types are supported at runtime level as long as the type is exactly the same. The supported types set only depends on what types are supported at planner level.

Reply:
My apology.  I mistook the way hive source to acquire the data. It seems that we can support all types that hive table spec can describe (in string) and can be converted to the runtime type, but need to enumerate these types in Hive source.

Reply:
Seems that we can't support TIMESTAMP_WITH_TIME_ZONE at present. I'm removing it from the supported types.